llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -3547,6 +3547,25 @@ const SCEV *ScalarEvolution::getUDivExpr(const SCEV *LHS,
     }
   }
 
+  // ((-C + (C smax %x)) /u %x) evaluates to zero, for any positive %x and any
+  // positive constant C.
+  if (const auto *AE = dyn_cast<SCEVAddExpr>(LHS);
+      AE && AE->getNumOperands() == 2) {
+    if (const auto *VC = dyn_cast<SCEVConstant>(AE->getOperand(0))) {
+      APInt NegC = VC->getAPInt();
+      if (NegC.isNegative() &&
+          NegC != APInt::getSignedMinValue(NegC.getBitWidth())) {
```

Comment:
```suggestion
      if ((-NegC).isStrictlyPositive()) {
```
I might phrase it like this, to align more closely with the comment/proof. But your version is fine as well.

Reply:
Exceptionally left mine as I deem it as slightly more intuitive.